openage/convert/texture.py

```diff
@@ -134,16 +146,36 @@ def __init__(self, input_data, palette=None, custom_cutter=None):
         self.image_data, (self.width, self.height), self.image_metadata\
             = merge_frames(frames)
 
-    def _slp_to_subtextures(self, frame, palette=None, custom_cutter=None):
+    def _slp_to_subtextures(self, frame, main_palette, player_palette=None,
+                            custom_cutter=None):
+        """
+        convert slp to subtexture or subtextures, using a palette.
+        """
+        # TODO this needs some _serious_ performance work
+        # (at least a 10x improvement, 50x would be better).
+        # ideas: remove PIL and use libpng via CPPInterface
+        subtex = TextureImage(
+            frame.get_picture_data(main_palette, player_palette,
+                                   self.player_id),
+            hotspot=frame.get_hotspot()
+        )
+
+        if custom_cutter:
+            # this may cut the texture into some parts
+            return custom_cutter.cut(subtex)
+        else:
+            return [subtex]
+
+    def _smp_to_subtextures(self, frame, main_palette, player_palette=None,
+                            custom_cutter=None):
         """
-        convert slp to subtexture or subtextures, use a palette.
+        convert smp to subtexture or subtextures, using a palette.
         """
         # TODO this needs some _serious_ performance work
         # (at least a 10x improvement, 50x would be better).
-        # ideas: remove PIL and use libpng via CPPInterface,
-        #        cythonize parts of SLP.py
+        # ideas: remove PIL and use libpng via CPPInterface
         subtex = TextureImage(
-            frame.get_picture_data(palette, self.player_id),
+            frame.get_picture_data(main_palette, player_palette),
```

Comment:
here we have to do the merging of the outline info and the graphics i think.
what do we do about the damage graphics? (we should likely discuss that in the chat :)